futures/tests/async_await_macros.rs

```diff
@@ -197,3 +200,91 @@ fn try_join_doesnt_require_unpin() {
         try_join!(x, y)
     };
 }
+
+#[test]
+fn test_manual_reset_event() {
+    let ev = manual_reset_event(false);
+    let ev2 = manual_reset_event(false);
+
+    let ev_set = ev.clone();
+    thread::spawn(move || {
+        thread::sleep(time::Duration::from_millis(100));
+        ev_set.set();
+    });
+
+    block_on(async {
+        let wait1 = ev.poll_set();
+        let wait2 = ev2.poll_set();
+
+        let mut nr_selects = 0;
+
+        // Since the futures from poll_set are Unpin, they must be alias by a reference which is
+        // !Unpin for select
```

Comment:
The inverse, `poll_set` returns an `impl Future + !Unpin` which `pin_mut!()` allows you to treat as an `impl Future + Unpin` by pre-pinning it to the current stack.

Reply:
Thanks for catching it, I got confused while writing this. Changed it.